docs/content/docs/how-to/mmv1-resource-documentation.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Add and update MMv1 resource documentation"
+summary: "Generated resources have generated documentation. This page describes the generation process and what YAML inputs are used."
+weight: 13
+---
+
+# MMv1 resource documentation
+
+A majority of the provider's documentation is generated using the same information that's used for generating the provider's Go code. For example, when adding a new field to a resource in the relevant `api.yaml` file we include a `description` field. This is used to set a field description in the resource's schema and is also used to document the field in generated markdown files.
+
+## Updating an existing MMv1 resource's documentation
+
+As a result of code generation, you often do not need to explicitly think about making updates to documentation. However it is a good idea to check the markdown changes when you generate the provider, especially if you are making lots of changes.
+
+## Adding new documentation for a new MMv1 resource
+
+More thought is required when adding a new resource from scratch. See the relevant section below for guidance about what YAML fields are needed to create complete documentation.
+
+## How generated documentation is made
+
+The Magic Modules compiler parses YAML files, creates Ruby objects that are populated with the data from those files, and then uses that data within template files to create produce the final markdown files.
+
+The main template used for documentation is `mmv1/templates/terraform/resource.html.markdown.erb`. As an example of how it works, you can see that the main title of documentation pages are created by some [processing of name data](https://github.com/GoogleCloudPlatform/magic-modules/blob/a69f1150de76f2b2cd9d37faa6bd44c1fb8a460a/mmv1/templates/terraform/resource.html.markdown.erb#L41) for a given resource and then using Ruby string methods to [print an escaped version of the name](https://github.com/GoogleCloudPlatform/magic-modules/blob/a69f1150de76f2b2cd9d37faa6bd44c1fb8a460a/mmv1/templates/terraform/resource.html.markdown.erb#L58) into an H1 header.
+
+
+## What YAML fields are used directly in the documentation
+
+Almost all fields in the YAML influences the documentation generated for a given resource. Below are descriptions of fields that are directly referenced in the documentation templates, and multiple fields not described here are relevant to the eventual documentation's content.
```

Comment:
It may be worth emphasizing that most of these fields impact the resource implementation as well as the documentation. That's possible to infer right now- either with MM experience or reading between the lines of the documentation as written- but extra clarity is likely to help.

Reply:
@rileykarson Agreed - I've changed this in https://github.com/GoogleCloudPlatform/magic-modules/pull/7131/commits/b1f4e30c6fdb7b89aff47dc6faed1c43f632be7e , and it's briefly mentioned in the opening of the file.

This bit of the docs feels a bit 'off' to me partly because there are several fields where it's incredibly obvious what they do, because they're the fields edited when implementing changes in a resource. However there are other fields are obscure or specific to documentation and need highlighting but they get lost in the long list of fields.

Would it be better to organise these tables by the section of the docs they impact? That way it could be more useful for people who are trying to address a specific problem e.g. 'How can I add more import formats for this resource?'.

Do you think that would be better?

Reply:
You've addressed my feedback with the change already, thanks!

The best idea I've got would be splitting the docs only parts and the side effects, but that increases the complexity of the page. It does break up "I need to make a specific change to the docs" and "I want to understand why the docs changed with X property being added". I'm not sure if I'd recommend making that change- it's debatably better with the relatively low number of settings right now.

Reply:
Yeah agreed, in that case I'll go ahead and merge. Thanks!